src/main/java/org/reaktivity/nukleus/http_cache/internal/proxy/cache/DefaultCache.java

```diff
@@ -116,224 +97,99 @@ public DefaultCacheEntry get(
     public DefaultCacheEntry supply(
         int requestHash)
     {
-        DefaultCacheEntry defaultCacheEntry = cachedEntries.get(requestHash);
-        if (defaultCacheEntry == null)
-        {
-            defaultCacheEntry = new DefaultCacheEntry(
-                    this,
-                    requestHash,
-                    cachedRequestBufferPool,
-                    cachedResponseBufferPool);
-            updateCache(requestHash, defaultCacheEntry);
-        }
-
-        return defaultCacheEntry;
+        return cachedEntries.computeIfAbsent(requestHash, this::newCacheEntry);
     }
 
-    public void signalForCacheEntry(
-        HttpCacheProxyCacheableRequest request,
-        long signalId)
-    {
-            writer.doSignal(request.signaler,
-                            request.acceptRouteId,
-                            request.acceptReplyId,
-                            supplyTrace.getAsLong(),
-                            signalId);
-    }
-
-    public void signalForUpdatedCacheEntry(int requestHash)
-    {
-       this.sendSignalToPendingInitialRequestsSubscribers(requestHash, CACHE_ENTRY_UPDATED_SIGNAL);
-    }
-
-    public void signalAbortAllSubscribers(
-        int requestHash)
+    public boolean matchCacheableResponse(
+        int requestHash,
+        String newEtag)
     {
-        this.sendSignalToPendingInitialRequestsSubscribers(requestHash, ABORT_SIGNAL);
+        DefaultCacheEntry cacheEntry = cachedEntries.get(requestHash);
+        return cacheEntry != null &&
+               cacheEntry.etag() != null &&
+               cacheEntry.etag().equals(newEtag) &&
+               cacheEntry.recentAuthorizationHeader() != null;
     }
 
-    public boolean handleCacheableRequest(
+    public boolean matchCacheableRequest(
         ListFW<HttpHeaderFW> requestHeaders,
         short authScope,
-        HttpCacheProxyCacheableRequest request)
+        int requestHash)
     {
         boolean canHandleRequest = false;
```

Comment:
Do we still need `canHandleRequest` variable?

Reply:
nope.Done.